src/components/MovieForm/MovieForm.test.js

```diff
@@ -0,0 +1,202 @@
+
+        });
+        movieData = {
+            id: 1,
+            movieName: "Interstellar",
+            movieGenre: "Sci-Fi",
+            movieDirector: "Christopher Nolan",
+            movieReleaseDate: new Date(),
+            movieImage: "",
+            movieDescription: "Very cool movie!"
+        }
+    });
+
+    it("should match snapshot", async () => {
+        const {container, getByText} = render(
+            <MemoryRouter>
+                <MovieForm />
+            </MemoryRouter>
+        );
+
+        await waitForElement(() => getByText("Genre"));
+
+        expect(container).toMatchSnapshot();
+    });
+
+    it("should post movie and get response", async () => {
+        const {getByLabelText, getByPlaceholderText, getByText} = render(
+            <MemoryRouter>
+                <MovieForm />
+            </MemoryRouter>
+        );
+
+        fireEvent.change(getByLabelText("Movie title"), {
+            target: {value: movieData.movieName}
+        });
+        fireEvent.change(getByLabelText("Director"), {
+            target: {value: movieData.movieDirector}
+        });
+        fireEvent.change(getByLabelText("Genre"), {
+            target: {value: movieData.movieGenre}
+        });
+        fireEvent.change(getByPlaceholderText("2014"), {
+            target: { value: movieData.movieReleaseDate}
+        });
+        fireEvent.change(getByLabelText("Description"), {
+            target: {value: movieData.movieDescription}
+        });
+
+        fireEvent.click(getByText("Add movie"));
+
+        await waitForElement(() => getByText("Movie added successfully.", {exact: false}));
+
+        expect(getByText("Movie added successfully", {exact: false})).toBeInTheDocument();
+    });
+
+    it("should render error on api fail", async () => {
+        postFail = true;
+        const {getByLabelText, getByPlaceholderText, getByText} = render(
+            <MemoryRouter>
+                <MovieForm />
+            </MemoryRouter>
+        );
+
+        fireEvent.change(getByLabelText("Movie title"), {
+            target: {value: movieData.movieName}
+        });
+        fireEvent.change(getByLabelText("Director"), {
+            target: {value: movieData.movieDirector}
+        });
+        fireEvent.change(getByLabelText("Genre"), {
+            target: {value: movieData.movieGenre}
+        });
+        fireEvent.change(getByPlaceholderText("2014"), {
+            target: { value: movieData.movieReleaseDate}
+        });
+        fireEvent.change(getByLabelText("Description"), {
+            target: {value: movieData.movieDescription}
+        });
+
+        fireEvent.click(getByText("Add movie"));
+
+        await waitForElement(() => getByText("Something went wrong while adding your movie."));
+
+        expect(getByText("Something went wrong while adding your movie.")).toBeInTheDocument();
+    });
+});
+
+describe("MovieForm GET & PUT", () => {
+    let getFail;
+    let putFail;
```

Comment:
W sumie to nigdzie później nie deklarujesz wartości tych zmiennych, a fetch się na nich opiera, także nie wiem jak to Ci działa xd ale jak działa to spoko xd